CONTRIBUTING.md

```diff
@@ -0,0 +1,200 @@
+
+## Working on the code
+
+### Fork the project
+
+In order to work on the project you will need your own fork. To do this click the "Fork" button on
+this project.
+
+Once the project is forked clone it to your local machine:
+
+```sh
+git clone git@github.com:your-user-name/cargo-leet
+cd cargo-leet
+git remote add upstream git@github.com:rust-practice/cargo-leet.git
+```
+
+This creates the directory cargo-leet and connects your repository to the upstream (main project) repository.
+
+### Creating a branch
+
+You want your main branch to reflect only production-ready code, so create a feature branch for
+making your changes. For example:
+
+```sh
+git checkout -b my-new-feature
+```
+
+This changes your working directory to the my-new-feature branch. Keep any changes in this branch
+specific to one bug or feature so the purpose is clear. You can have many my-new-features and switch
+in between them using the git checkout command.
+
+When creating this branch, make sure your main branch is up to date with the latest upstream
+main version. To update your local main branch, you can do:
+
+```sh
+git checkout main
+git pull upstream main --ff-only
+```
+
+### Code linting, formatting, and tests
+
+You can run linting on your code at any time with:
+
+```sh
+cargo clippy
+```
+
+To format the code run:
+
+```sh
+cargo fmt
+```
+
+To run the tests:
+
+```sh
+cargo test
+```
+
+To ensure the code compiles run:
+
+```sh
+cargo check
+```
+
+Be sure to run all these checks before submitting your pull request.
+
+## Committing your code
+
+Once you have made changes to the code on your branch you can see which files have changed by running:
+
+```sh
+git status
+```
+
+If new files were created that and are not tracked by git they can be added by running:
+
+```sh
+git add .
+```
+
+Now you can commit your changes in your local repository:
+
+```sh
+git commit -am 'Some short helpful message to describe your changes'
+```
+
+## Push your changes
+
+Once your changes are ready and all linting/tests are passing you can push your changes to your forked repositry:
+
+```sh
+git push origin my-new-feature
+```
+
+origin is the default name of your remote repositry on GitHub. You can see all of your remote repositories by running:
+
+```sh
+git remote -v
+```
+
+## Making a Pull Request
+
+After pushing your code to origin it is now on GitHub but not yet part of the cargo-leet project.
+When you’re ready to ask for a code review, file a pull request. Before you do, once again make sure
+that you have followed all the guidelines outlined in this document regarding code style, tests, and
+documentation. You should also double check your branch changes against the branch it was based on by:
+
+1. Navigating to your repository on GitHub
+1. Click on Branches
+1. Click on the Compare button for your feature branch
+1. Select the base and compare branches, if necessary. This will be main and my-new-feature, respectively.
```

Comment:
I think it might be better for them to make pull requests against develop (which will be the default based on how we have the repo setup)